### ISSUE TITLE
Decouple the code for explore(obj) and explore() ?

### ISSUE THREAD

**Issue by nthiery** [sagemath/sage-explorer#12]
The layout of the widgets for `explore()` and `explore(obj)` may become relatively different. Maybe we would want to split the code accordingly into two classes? Or is there a lot shared between the two?

**Comment by zerline** [comment]
I agree, they may become rather different. There is still a lot to be shared: the display structure (unless we want it more different than now). And a few methods. A possibility could be to write a base class (XYZExplorer), then make 2 classes ObjectExplorer and ExplorerIndex (or any names) inherit from the base class.

**Comment by nthiery** [comment]
>    I agree, they may become rather different. There is still a lot to be
>    shared: the display structure (unless we want it more different than
>    now). And a few methods.

And the history management.

>    A possibility could be to write a base class (XYZExplorer), then
>    make 2 classes ObjectExplorer and ExplorerIndex (or any names)
>    inherit from the base class.

+1

It could be:

- Explorer
- ObjectExplorer
- SageExplorer

No rush though, unless this is of immediate help to maintain / extend
the code.

**Comment by zerline** [comment]
It will be ..

**Comment by zerline** [comment]
With a4c38c51dcd32b13, we decided to use the same display.
In the future, we would like to display, for a catalog, all its values as a table with name, icon, description and args box in case args are required.
